specification/containerservice/resource-manager/Microsoft.ContainerService/stable/2020-11-01/managedClusters.json

```diff
@@ -2828,6 +2832,93 @@
       ],
       "description": "A Kubernetes add-on profile for a managed cluster."
     },
+    "ManagedClusterPodIdentity": {
+      "properties": {
+        "name": {
+          "type": "string",
+          "description": "Name of the pod identity."
+        },
+        "namespace": {
+          "type": "string",
+          "description": "Namespace of the pod identity."
+        },
+        "identity": {
+          "$ref": "#/definitions/UserAssignedIdentity",
+          "description": "Information of the user assigned identity."
+        },
+        "provisioningState": {
+          "type": "string",
+          "readOnly": true,
+          "description": "The current provisioning state of the pod identity.",
+          "enum": [
+            "Assigned",
+            "Updating",
+            "Deleting",
+            "Failed"
+          ],
+          "x-ms-enum": {
+            "name": "ManagedClusterPodIdentityProvisioningState",
+            "modelAsString": true
+          }
+        },
+        "provisioningInfo": {
```

Comment:
just curious, why put this as a property, from description it should be output of an API call right?

Reply:
would like to involve ARM review since this PR introduce a property ProvisioningInfo to indicate last assignment error, not very common. Invite ARM review to share more insight on this.

Reply:
This sort of things comes up with RPs that provision additional resources during CRUD operations on their own resource types. If there's any possibility of failure, you typically have to expose the errors for troubleshooting purposes. As such, it's generally allowed and we don't have a standard contract for it.

In this specific instance, my suggestion would be to name it `provisioningStatus` instead of `provisioningInfo`, but I'll leave the choice to the RP team.

---
In reply to: [511770579](https://github.com/Azure/azure-rest-api-specs/pull/11366#discussion_r511770579) [](ancestors = 511770579)

Reply:
Thanks @majastrz and @yungezz for the reviewing! For current changes, I will leave the `provisioningInfo` as is.